modules/compute/gke-node-pool/gpu_direct.tf

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+# Enable GPUDirect for A3 and A3Mega VMs, this involve multiple kubectl steps to integrate with the created cluster
+# 1. Install NCCL plugin daemonset
+# 2. Install NRI plugin daemonset
+# 3. Update user workload to inject rxdm sidecar and other required annotation, volume etc.
+locals {
+  user_workload_path_tcpx  = var.user_workload_path == null ? "${path.module}/gpu-direct-workload/sample-tcpx-workload-job.yaml" : "${path.module}/${var.user_workload_path}"
```

Comment:
This assumes that the user supplied the path to a file that lives within the module. This assumption is likely not to be true. Instead we should just treat input as a normal path and then use ghpc_stage in the blueprint to make sure the files get staged to the correct location. 

Here are the docs for [ghpc_stage](https://developer.hashicorp.com/terraform/language/v1.5.x/checks).  
Here is an [example of it in use](https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/1e38ce0c2353a3b14520043ab27514d8defd2d71/community/examples/htc-slurm.yaml#L35).

Reply:
this is no longer applicable because we are now getting user instruction to run the script to update their own workload instead. In the future if we move the logic out to its own module, we might want to provide more control to it